model/sale.py

```diff
@@ -1,33 +1,44 @@
 from model import Book
 from datetime import datetime
-
+from dao import BookDao
 
 class Sale:
     """Modelo de Venda"""
 
     __attributes = ('_title_book', '_quantity_book')
 
-    def __init__(self, title_book: Book, quantity_book: int) -> None:
-        self._title_book = title_book
-        self.quantity_book = quantity_book
+    book_dao = BookDao()
+
+    def __init__(self, title_book: str, quantity_book: int) -> None:
+        self.__set_book_for_sale(title_book)
+        self.quantity_sale = quantity_book
+        self._registration_time = datetime.now()
```

Comment:
Aqui não entendi por que você quebrou o objeto book, pra guardar só o título, qual é o benefício disso?

Reply:
Essa foi uma dúvida que tive antes, se a venda recebe apenas o título do livro ou o livro em si. Não vejo sentido dela receber o livro em si, porque na instância do livro tem coisas internas dele, como por exemplo: preço, autor, categoria... Então fez sentido o usuário dizer: "eu quero o livro Fluente Python", ou seja você passa o título do livro para uma venda. Só que quando eu passo esse título, tenho que ver se há esse título no acervo, faço isso com o find_by_title e guardo o objeto completo na venda.

Reply:
Certo, o usuário pode dizer isso "Quero o livro Python Fluente", Na busca, você pode ter 2 resultados, digamos que há duas edições, ai você adiciona na venda e vai fazer o checkout, o vendedor quer conferir pro usuário se selecionou o livro correto. Ele consegue fazer isso apenas olhando o título e o preço? Se a resposta for sim, beleza :)

Reply:
No ```find_by_title``` ele retornará apenas um resultado conforme o título passado. Não há essa possibilidade dele retornar mais de um resultado nessa função.

Já na função ```find_all_books_with_title``` ele pode retornar mais de um, então se no meu catálogo tiver:
 - Python Fluente v1
 - Python Fluente Básico
 - Python Fluente v2

E eu procurar pelo título Python Fluente, essa função irá filtrar todos os livros que possuem esse título, daí retornará os 3. 

> Porque eu fiz duas funções?
Porque no  ```find_by_title``` eu quero que retorne apenas o livro exato que o usuário me disser na venda e caso não tenha, uma exceção irá ocorrer.

Já no  ```find_all_books_with_title``` eu quero todas as ocorrências daquele título.

Reply:
Preciso alterar algo mais aqui? Não entendi se devo fazer mais modificações

Reply:
Não precisa, acho que foram bons pontos, apenas por questão de expressividade, acho que faz sentido usar no nome algo que indique o resultado da chamada da função, algo no sentido: findOne e find ou findMany algo assim.